eox_core/logging.py

```diff
@@ -0,0 +1,51 @@
+"""
+The logging module defines functions that are used for the logging of the
+pipeline steps.
+"""
+import logging
+import sys
+from pprint import pformat
+
+LOG = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument,keyword-arg-before-vararg
+def logging_pipeline_step(level, exception_message, **local_vars):
+    """
+    This function logs and exception with the information from the pipeline step that failed.
+    If the logLevel setting is in DEBUG mode then logs all the arguments passed to the step.
+
+    For example:
+        To set the verbosity level to DEBUG add the setting:
+
+        "BACKEND_OPTIONS": { "logLevel":"DEBUG" }
+    """
+    user = local_vars.get("user")
+    backend = local_vars.get("backend")
+
+    message = "PIPELINE-STEP:{step} - USER:{user} - BACKEND:{backend_name} - REDIRECT_URI:{uri} - MESSAGE:{msg}".format(
+        step=sys._getframe(1).f_code.co_name,  # pylint: disable=protected-access
+        user=user,
+        backend_name=getattr(backend, "name", ""),
+        uri=getattr(backend, "redirect_uri", ""),
+        msg=exception_message,
+    )
+
+    if backend and backend.setting("BACKEND_OPTIONS", {}).get("logLevel", "").lower() == "debug":
+        div = "=" * 100
+        details = local_vars.get("details")
+        args = local_vars.get("args", [])
+        kwargs = local_vars.get("kwargs", {})
```

Comment:
There are a couple of pipeline steps that have as argument: `is_new`, so for these cases it will not be reflected in the kwargs.

Maybe we can remove that from the arguments of these pipeline steps and put at the beginning of each step

is_new = kwargs.get('is_new')

What do you think @magajh ?

Reply:
@morenol  Thank you for noticing, I missed that argument. Yes, I think that would be a good solution